### PR TITLE
fix: validate device index before using it to calculate offset

### PR DESCRIPTION
**Pull request by jkralik**

### CODE REVIEW

File: api/oc_core_res.c [hunk 1]
```diff
@@ -589,6 +589,9 @@ core_get_resource_memory_by_index(int type, size_t device)
   if (type < OCF_CON) {
     return &g_core_resources[type];
   }
+  if (device >= OC_ATOMIC_LOAD32(g_device_count)) {
```
Comment (Danielius1922) [inline api/oc_core_res.c:592]: add line to CoreGetResourceByIndex_F to test with invalid device index

File: api/oc_core_res.c [hunk 2]
```diff
@@ -669,6 +672,9 @@ oc_core_is_SVR(const oc_resource_t *resource, size_t device)
   if (resource == NULL) {
     return false;
   }
+  if (device >= OC_ATOMIC_LOAD32(g_device_count)) {
+    return NULL;
```
Comment (Danielius1922) [inline api/oc_core_res.c:676]: `NULL` -> `false`

Add call with invalid device index to the beginning of `CoreGetResourceIsSVR_P`

File: api/oc_core_res.c [hunk 3]
```diff
@@ -695,6 +701,10 @@ oc_core_is_vertical_resource(const oc_resource_t *resource, size_t device)
     }
   }
 
+  if (device >= OC_ATOMIC_LOAD32(g_device_count)) {
+    return false;
```
Comment (Danielius1922) [inline api/oc_core_res.c:705]: add call to `CoreGetResourceIsVerticalResource_P`

File: api/oc_core_res.c [hunk 4]
```diff
@@ -720,6 +730,10 @@ oc_core_is_DCR(const oc_resource_t *resource, size_t device)
...
```
Comment (Danielius1922) [inline api/oc_core_res.c:734]: add call to CoreGetResourceIsDCR_P

File: api/oc_core_res.c [hunk 5]
```diff
@@ -866,6 +880,9 @@ oc_core_get_resource_by_uri(const char *uri, size_t device)
   if (type < OCF_CON) {
     return &g_core_resources[type];
   }
+  if (device >= OC_ATOMIC_LOAD32(g_device_count)) {
+    return NULL;
```
Comment (Danielius1922) [inline api/oc_core_res.c:884]:
```C
TEST_F(TestCoreResourceWithDevice, CoreGetResourceByURI_F)
{
  EXPECT_EQ(nullptr, oc_core_get_resource_by_uri("", 0));
  EXPECT_EQ(nullptr, oc_core_get_resource_by_uri("/oic/d", static_cast<size_t>(-1)));
}
```

File: api/oc_core_res.c (hunk 4 above)
Comment (jkralik) [reply api/oc_core_res.c:734]: added

File: api/oc_core_res.c (hunk 3 above)
Comment (jkralik) [reply api/oc_core_res.c:705]: added

File: api/oc_core_res.c (hunk 1 above)
Comment (jkralik) [reply api/oc_core_res.c:592]: added

File: api/oc_core_res.c (hunk 2 above)
Comment (jkralik) [reply api/oc_core_res.c]: Fixed

File: api/oc_core_res.c (hunk 5 above)
Comment (jkralik) [reply api/oc_core_res.c:893]: Added

**Review by Danielius1922 — APPROVED** [review]

**Pull request merged**